src/components/SQForm/SQForm.tsx

```diff
@@ -1,25 +1,55 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import Grid from '@material-ui/core/Grid';
 import {Formik, Form} from 'formik';
 import {useDebouncedCallback} from 'use-debounce';
 import {object as YupObject, setLocale} from 'yup';
 import {useInitialRequiredErrors} from '../../hooks/useInitialRequiredErrors';
+import type {GridProps} from '@material-ui/core';
+import type {FormikHelpers, FormikValues} from 'formik';
+import type {ObjectShape} from 'yup/lib/object';
 
 setLocale({
   mixed: {
     required: 'Required',
   },
 });
 
-function SQForm({
+interface SQFormProps<Values> {
+  /** Form Input(s) */
+  children: React.ReactNode;
+  /** Bool to pass through to Formik. https://formik.org/docs/api/formik#enablereinitialize-boolean */
+  enableReinitialize?: boolean;
+  /** Form Entity Object */
+  initialValues: Record<string, Values>;
```

Comment:
The type here would just be `Values` right?  I think we might want to do `Values extends Record<string, unknown>` but unsure. `Values` will definitely need to be a type like `{someKey: string, anotherKey: number}` where the keys are names of fields in the form. Basically initialValues. We can enforce that with `extends`

Reply:
Let me look into this.  I was having some issues with the stories not liking `Values` here since it could not guarantee what it was passing was of type `Values` but let me investigate further.

Reply:
I could see a case where we're trying to index it or use it like an object and TS can't be guaranteed that `Values` is a type that allows for indexing or being used like an object. If oyu do `Value extends Record<string, unknown>` then TS won't allow a generic to be passed that isn't an object. I'd be interested to know what the issue is if it isn't that.

Reply:
I'll try that and see where I can get.

Reply:
@20BBrown14 After trying to get this to work using your idea, I am still having errors, and am not sure how to resolve them using your suggested implementation.  Here is a loom that shows the issue I am running into.  If you or anyone has ideas of how to get around this let me know, thanks.
https://www.loom.com/share/41e6d770d38d4dd489eb5382a2a88ec4.

Reply:
@aharpt thanks for giving it a go. After look at your loom I tried some stuff out and landed on this solution that I've described in this loom https://www.loom.com/share/fbee45c8f27446679bf05b9d5151b738

Reply:
@20BBrown14 Very cool!  I have made the changes.  Thanks for the help!